src/lib/helpers/BoxHelper.svelte

```diff
@@ -0,0 +1,23 @@
+<script lang="ts">
+    import { type ColorRepresentation, BoxHelper } from 'three'
+    import Object3DInstance from '../instances/Object3DInstance.svelte'
+    import { getParent } from '../internal/HierarchicalObject.svelte'
+  
+    //self
+    export let color:ColorRepresentation = undefined;
+  
+    const parent = getParent()
+  
+    let helper: BoxHelper
+  
+    if (!($parent instanceof Object3DInstance)) {
+      console.warn('<BoxHelper> can only be used as a direct child of <Mesh>') //This description is somewhat innacurate and needs to be improved.
+    } else {
+      helper = new BoxHelper($parent,color)

```

Comment:
This helper poses an interesting problem as it's not auto-updating to match the parent it is trying to visualise.
The [docs](https://threejs.org/docs/index.html?q=boxhel#api/en/helpers/BoxHelper) state you need to call `update()` to ~'update the helper's geometry to match the dimensions of the object, including any children.'.
As this is something that is hard to track with threltes unified render loop and i oppose the use of `useFrame` for a debugging utility I suggest to export the function update. I'd refactor it like this:
```ts
const parent = getParent()
// https://github.com/mrdoob/three.js/blob/860af3c012915be534f8a5c37d745e2dff339e04/src/math/Box3.js#L191 I don't think it's throwing for objects that are not a Mesh, so we don't do that too.
export const boxHelper = new BoxHelper($parent, color) // remember to rename this to ThreeBoxHelper as stated in ArrowHelper comment
export const update = () => boxHelper.update()
```
This way users can update the helper easily by themselves until we find a better approach.